src/app/app-routing.module.ts

```diff
@@ -5,14 +5,42 @@ import {CoursesSectionComponent} from './features/course/courses-section/courses
 import {PageNotFoundComponent} from './shared/page-not-found/page-not-found.component';
 import {UserLoginComponent} from './features/login/user-login/user-login.component';
 import {AddCourseWindowComponent} from './features/add-course/add-course-window/add-course-window.component';
+import {AuthGuard} from './core/guards/auth.guard';
 
 const routes: Routes = [
   {path: '', redirectTo: 'courses', pathMatch: 'full'},
 
-  {path: 'courses', component: CoursesSectionComponent},
-  {path: 'login', component: UserLoginComponent},
-  {path: '404', component: PageNotFoundComponent},
-  {path: 'add-course', component: AddCourseWindowComponent},
+  {
+    path: 'login',
+    component: UserLoginComponent,
```

Comment:
Use import module for true lazy loading
https://angular.io/guide/router#lazy-loading-route-configuration